File: Hackerrank/Go/plus-minus.go
// Author: Vatsal Chanana
// URL: https://www.hackerrank.com/challenges/plus-minus/problem

package main

import (
    "bufio"
    "fmt"
    "io"
    "os"
    "strconv"
    "strings"
)

// Complete the plusMinus function below.
func plusMinus(arr []int32) {
    pos, neg, zero := 0, 0, 0

    for _, num := range arr {
        if num > 0 {
            pos++
        } else if num == 0 {
            zero++
        } else {
            neg++
        }
    }

    size := float32(len(arr))
    fmt.Println(float32(pos) / size)
    fmt.Println(float32(neg) / size)
    fmt.Println(float32(zero) / size)
}

func main() {
    reader := bufio.NewReaderSize(os.Stdin, 1024 * 1024)

    nTemp, err := strconv.ParseInt(readLine(reader), 10, 64)
    checkError(err)
    n := int32(nTemp)

    arrTemp := strings.Split(readLine(reader), " ")

    var arr []int32

    for i := 0; i < int(n); i++ {
        arrItemTemp, err := strconv.ParseInt(arrTemp[i], 10, 64)
        checkError(err)
        arrItem := int32(arrItemTemp)
        arr = append(arr, arrItem)
    }

    plusMinus(arr)
}

func readLine(reader *bufio.Reader) string {
    str, _, err := reader.ReadLine()
    if err == io.EOF {
        return ""
    }

    return strings.TrimRight(string(str), "\r\n")
}

func checkError(err error) {
    if err != nil {
        panic(err)
    }
}

File: Hackerrank/Python/plus-minus.py
#This code is contributed by Shreevatsa N, @vatsa287
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the plusMinus function below.
def plusMinus(arr):
    length = len(arr)
    cpos , cneg ,czer = 0, 0, 0
    for i in range(0,length):
        if arr[i] > 0:		#counters for pos,neg and zero nus.
            cpos+=1
        elif arr[i] < 0:
            cneg+=1
        else:
            czer+=1
    print(cpos/length)		#no of pos/neg/zero nos divided by total nos.
    print(cneg/length)
    print(czer/length)
                    


if __name__ == '__main__':
    n = int(input())

    arr = list(map(int, input().rstrip().split()))

    plusMinus(arr)
